Can use options api same as stylelint

gucong3000 opened the pull request:
Try load config file use `stylelint`. That I can use one config object when I use stylelint in same time.

```
let stylelintrc = {
	//
}

let processors = [
	require("stylefmt")(stylelintrc),
	require("stylelint")(stylelintrc),
];

postcss(processors)
```

morishitter commented on lib/params.js:
```diff
@@ -119,7 +106,23 @@ function params (options) {
+function loadStylelintConfig (options) {
+  // same as the options in stylelint. see: http://stylelint.io/user-guide/node-api/#options
+  try {
+    return require('stylelint/dist/buildConfig').default(options).catch(function() {
```
I'd not like to `require('stylelint/dist/buildConfig')` directly.

gucong3000 replied:
There is a very similar code here:
https://github.com/spalger/gulp-jshint/blob/master/src/lint.js#L4

morishitter commented on lib/params.js:
```diff
@@ -21,9 +21,9 @@ function hasScssInWorkingDir (wd) {
   var config
   if (hasScss) {
-    config = editorconfig.parseSync(path.resolve(process.cwd(), '*.scss'))
+    config = editorconfig.parseSync(path.resolve(from, '*.scss'))
```
I think, if `from` is `undefined`, it should be used `process.cwd()`.

gucong3000 replied:
`getIndentationFromEditorConfig()` is only call by `params()`, `process.cwd()` will be pass.

gucong3000 replied:
I think we should use real file path to find `.editorconfig`. I will commit new code.

morishitter commented:
@gucong3000 On second thought, I couldn't agree your PR. If stylelint provides api to load its configuration (e.g. `stylelint.loadConfig`), I'll try to read its config, and `options` argument in `params` function is for stylefmt, not for stylelint. What do you think?

gucong3000 commented:
I think we should persuade them to provide API

gucong3000 commented:
`require('stylelint/dist/buildConfig')`is removed.

gucong3000 commented:
# CHANGELOG

- Fixed: `config` option to CLI. Async with [stylelint CLI](https://github.com/stylelint/stylelint/blob/master/src/cli.js#L44)
  - support name of a node_module
  - support absolute path
  - support relative path relative to `process.cwd()`.
- Added: `--config-basedir` option to CLI. Async with [stylelint CLI](https://github.com/stylelint/stylelint/blob/master/src/cli.js#L57)
- Fixed: `config` option to Node.js API. Async with [stylelint API](http://stylelint.io/user-guide/node-api/#config)
- Added: `configBasedir` option to Node.js API. Async with [stylelint API](http://stylelint.io/user-guide/node-api/#configbasedir)
- Added: `configFile` option to Node.js API. Async with [stylelint API](http://stylelint.io/user-guide/node-api/#configfile)
- Fixed: real style file's path will pass to `editorconfig`

gucong3000 commented:
@morishitter I was removed `require('stylelint/dist/buildConfig')`.
all option to Node.js API and CLI was same to `stylelint`

gucong3000 commented:
All the changes you don't want have been removed.
Other changes have been organized as [CHANGELOG](#issuecomment-244682446)

I really need these interfaces to customize the path of the configuration file or configuration object.
If we do not have these interfaces, `stylefmt` will not be able to be integrated with IDE, because `stylefmt` always read the configuration file from the `process.cwd()`, it will be the directory where the IDE, rather than the project directory
